### DISCUSSION TITLE
How to validate a configuration file with the built-in CLI validator?

### DISCUSSION THREAD

**Discussion by kevinmatthes** [renovatebot/renovate#27202] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.181.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

Renovate also exists as a CLI to be installed using ``npm``.  This allows for calling the built-in validator on a project's Renovate configuration within a GitHub CI to check for typos and validity.  So far, it technically sufficed to run the following command to check a certain configuration file, say ``.renovaterc.json5``:

```bash
npm install renovate && renovate-config-validator .renovaterc.json5 --strict
```

There are also some GitHub Actions which encapsulate this instruction and provide some further utilities but basically, this simple line suffices for the most use cases.

Some hours ago, a new Renovate version series, 37.181.x, was published and now, the described instruction does not work, anymore.  I assume that the calling semantics have changed because neither the command ``renovate-config-validator`` nor ``renovate`` can be found.  For reproduction, it suffices to put the above line into some GitHub CI workflow.

My question is:  how do I need to call the Renovate CLI to check a given Renovate configuration file with the built-in validator?

**Comment by adamnolte** [comment]
I think this is related to https://github.com/renovatebot/renovate/discussions/27200

**Reply by kevinmatthes** [reply, marked as answer]
Thank you for comment!  This was indeed the reason why there was some trouble calling the Renovate CLI.  Now that the bug is fixed, the following command will validate any Renovate configuration file, again with ``.renovaterc.json5`` being used as concrete example:

```bash
npx -p renovate -c 'renovate-config-validator .renovaterc.json5 --strict' -y
```

This adjustment to the command I named in my original question is another requirement to make the CLI call work as expected.  This solution is also provided as a GitHub Action on its own:  https://github.com/suzuki-shunsuke/github-action-renovate-config-validator.
